invisible_cities/cities/detsim_simulate_electrons.py

```diff
@@ -0,0 +1,140 @@
+from typing import Tuple
+
+from invisible_cities.cities.detsim_get_psf import binedges_from_bincenters
+
+#######################################
+######### ELECTRON SIMULATION #########
+#######################################
+def generate_ionization_electrons(wi          : float,
+                                  fano_factor : float,
+                                  energies    : np.ndarray) -> np.ndarray:
+    """ generate ionization secondary electrons from energy deposits
+
+    Parrameters:
+        :wi: float
+            ionization yield
+        :fano_factor: float
+            fano-factor. related with the deviation in ionization electrons
+        :energies: np.ndarray
+            energy hits
+    Returns:
+        :nes: np.ndarray
+            the ionization electrons per hit
+    """
+    nes  = np.array(energies/wi, dtype = int)
+    pois = nes < 10
+    nes[ pois] = np.random.poisson(nes[pois])
+    nes[~pois] = np.round(np.random.normal(nes[~pois], np.sqrt(nes[~pois] * fano_factor)))
+    return nes
+
+
+def drift_electrons(lifetime       : float,
+                    drift_velocity : float,
+                    zs             : np.ndarray,
+                    electrons      : np.ndarray) -> np.array:
+    """ returns number of electrons due to lifetime loses from secondary electrons
+
+    Parameters:
+        :lifetime: float
+            electron lifetime
+        :drif_velocity: float
+            drif velocity at the active volume of the detector
+        :zs:
+            the z hits
+        :electrons:
+    Returns:
+        :nes: np.ndarray
+            the drifted electrons per initial hit
+    """
+    ts  = zs / drift_velocity
+    nes = electrons - np.random.poisson(electrons * (1. - np.exp(-ts/lifetime)))
+    nes[nes < 0] = 0
+    return nes
+
+
+def diffuse_electrons(transverse_diffusion   : float,
+                      longitudinal_diffusion : float,
+                      xs                     : np.ndarray,
+                      ys                     : np.ndarray,
+                      zs                     : np.ndarray,
+                      electrons              : np.ndarray)\
+                      -> Tuple[np.ndarray, np.ndarray, np.ndarray]:
+    """
+    starting from hits with positions xs, ys, zs, and number of electrons,
+    apply diffusion and return diffused positions xs, ys, zs for each electron.
+
+    Paramters:
+        :transverse_diffusion: float
+        :longitudinal_diffusion: float
+        :xs, ys, zs: np.ndarray (1D of size: #hits)
+            postions of initial hits
+        :electrons:
+            electrons at the EL before drifting per hit
+    Returns:
+        :dxs, dys, dzs: np.ndarray (1D of size: #hits x #total electrons)
+            diffused positions
+    """
+    xs = np.repeat(xs, electrons.astype(int))
+    ys = np.repeat(ys, electrons.astype(int))
+    zs = np.repeat(zs, electrons.astype(int))
+
+    # substitute z<0 to z=0
+    sel = zs<0
+    zs[sel] = 0
```

Comment:
I added this lines as a temporal fix because some nexus hits were negative (with very smal values of the order of -1e-14). If I am correct, @jmalbos said it was related with the precission of float, so could be dowgraded. Since this was still not done in nexus I kept the fix, which in fact is neccesary, since negative values were entering in a square root.